libgalois/test/reduce-error-info.cpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#include <functional>
+#include <iostream>
+
+#include "katana/Galois.h"
+#include "katana/ResultReduction.h"
+
+katana::Result<void>
+ErrorOnTrue(bool error) {
+  if (error) {
+    return KATANA_ERROR(katana::ErrorCode::NotFound, "error");
+  }
+  return katana::ResultSuccess();
+}
+
+katana::Result<bool>
+ErrorOnTrue(bool error, bool return_value) {
+  if (error) {
+    return KATANA_ERROR(katana::ErrorCode::NotFound, "error");
+  }
+  return return_value;
+}
+
+void
+TestVoidFunc() {
+  katana::GReduceCopyableErrorInfo error_info;
+  katana::do_all(katana::iterate(size_t{0}, size_t{100}), [&](size_t i) {
+    KATANA_REDUCTION_CHECKED(error_info, ErrorOnTrue((i % 10) == 0));
+    KATANA_LOG_ASSERT((i % 10) != 0);
+  });
+  std::optional<katana::CopyableErrorInfo> reduced_error_info =
```

Comment:
I would hide the `reduce()` API behind something easier, e.g. a conversion operator `operator CopyableErrorInfo() {...}`.

Reply:
It's not there in the test but the typical usage would be:
```cpp
  if (auto error = error_info.reduce()) {
    return *error;
  }
```
It would be great if there's a more concise way to write that code. I can change it to:
```cpp
  if (auto error = error_info()) {
    return *error;
  }
```
Is that what you're suggesting?

Reply:
Yes.